lib/rpmvs.c

```diff
@@ -566,7 +566,9 @@ static rpmRC verifyDigest(struct rpmsinfo_s *sinfo)
 static rpmRC
 verifySignature(rpmKeyring keyring, struct rpmsinfo_s *sinfo)
 {
-    rpmRC res = rpmKeyringVerifySig(keyring, sinfo->sig, sinfo->ctx);
+    rpmRC res = RPMRC_FAIL;
+    if (sinfo->sig && pgpSignatureType(sinfo->sig) == PGPSIGTYPE_BINARY)
```

Comment:
See below, handle NULL inside pgpSignatureType() instead which makes it nicer for the user.